File: Rust/benchmark.rs
mod fft;

use std::time::Instant;

const LOG2FFTSIZE: u32 = 12;
const FFT_REPEAT: u32 = 1000;

const SIZE: usize = (1<<LOG2FFTSIZE);

fn main() {
    let mut xy: [[f64; 2]; SIZE] = [[0.0; 2]; SIZE];
    let mut xy_out_fft = [[0.0; 2]; SIZE];

    for i in 0..SIZE/2    { xy[i] = [ 1.0, 0.0]; }
    for i in SIZE/2..SIZE { xy[i] = [-1.0, 0.0]; }

// FFT
    let start_time = Instant::now();
    for _i in 0..FFT_REPEAT { unsafe { fft::fft(LOG2FFTSIZE, xy, &mut xy_out_fft) } }
    let elapsed_time = start_time.elapsed();
    let milliseconds = (elapsed_time.as_secs() as f64 * 1000.0) +
		       (elapsed_time.subsec_nanos() as f64 / 1_000_000.0);

    println!("{} piece(s) of {} pt FFT;    {} ms/piece\n\n", FFT_REPEAT, SIZE, milliseconds/FFT_REPEAT as f64);

    for i in 0..6 {
	println!("{}  {} {}", i, xy_out_fft[i][0], xy_out_fft[i][1])
    }
}

File: Go-DFT-per-sample-double/benchmark.go
package main

import (
    "fmt"
    "time"
    "dft"
)

const LOG2FFTSIZE uint= 12
const FFT_REPEAT int = 10

const SIZE int = (1<<LOG2FFTSIZE)
var xy [SIZE]complex128
var xy_out_fft [SIZE]complex128

func main() {
    var i int
    for i=0; i<SIZE/2; i++ { xy[i]= (1.+0i) }
    for    ; i<SIZE  ; i++ { xy[i]= (-1.+0i) }

// FFT
    dft.Dft_init(SIZE)
    var startTime = time.Now().UnixNano()
    for i=0; i<FFT_REPEAT; i++ {
	for j:=0; j<SIZE; j++ {
	    dft.Dft_sample(&xy_out_fft, xy[j])
	}
    }
    var endTime = time.Now().UnixNano()

    fmt.Printf("%8d piece(s) of %d pt FFT;    %.3f ms/piece\n\n", FFT_REPEAT, SIZE, float64(endTime-startTime)/float64(1e6)/float64(FFT_REPEAT))

    for i=0; i<6; i++ {
	fmt.Println(i, xy_out_fft[i])
    }
}
